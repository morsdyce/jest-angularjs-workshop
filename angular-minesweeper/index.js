import 'common/assets/stylesheets/minefield.css';

import App from 'angular-minesweeper/components/app/app.module';
import Common from 'angular-minesweeper/common/common.module';
import Board from 'angular-minesweeper/components/board/board.module';

export default angular.module('ngReact', [
  Common.name,
  Board.name,
  App.name
]);