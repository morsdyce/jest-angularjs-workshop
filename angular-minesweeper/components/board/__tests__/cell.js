import 'angular-minesweeper';
import SPOT_TYPES from 'common/spot/spot-types.constants';

describe('<cell>', () => {
  let createComponent, $scope, element;

  beforeEach(angular.mock.module('ngReact.board'));

  beforeEach(inject(($rootScope, $compile) => {
    createComponent = (html, scopeAttrs) => {
      $scope = angular.extend($rootScope.$new(), scopeAttrs);

      element = angular.element(html);
      element = $compile(element)($scope);
      $scope.$digest();
    };
  }));

  test('should show an unrevealed spot', () => {
    const html = `<cell spot="spot"></cell>`;
    createComponent(html, {
      spot: {
        isRevealed: false,
        content: {
          type: SPOT_TYPES.MINE.type
        }
      }
    });

    expect(element.html()).toMatchSnapshot();
  });

  test('should show an empty spot', () => {
    const html = `<cell spot="spot"></cell>`;
    createComponent(html, {
      spot: {
        isRevealed: true,
        content: {
          type: SPOT_TYPES.EMPTY.type
        }
      }
    });

    expect(element.html()).toMatchSnapshot();
  });

  test('should show a mine spot', () => {
    const html = `<cell spot="spot"></cell>`;
    createComponent(html, {
      spot: {
        isRevealed: true,
        content: {
          type: SPOT_TYPES.MINE.type
        }
      }
    });

    expect(element.html()).toMatchSnapshot();
  });

  test('should show a flagged spot', () => {
    const html = `<cell spot="spot"></cell>`;
    createComponent(html, {
      spot: {
        isRevealed: false,
        isFlagged: true,
        content: {
          type: SPOT_TYPES.MINE.type
        }
      }
    });

    expect(element.html()).toMatchSnapshot();
  });

  Object.entries(SPOT_TYPES)
    .filter(([type]) => type.includes('NUMBER'))
    .forEach(([type, value]) => {
      test(`should show a ${ type } spot`, () => {
        const html = `<cell spot="spot"></cell>`;
        createComponent(html, {
          spot: {
            isRevealed: true,
            isFlagged: false,
            content: {
              type: value.type
            }
          }
        });

        expect(element.html()).toMatchSnapshot();
      });
    });



});

