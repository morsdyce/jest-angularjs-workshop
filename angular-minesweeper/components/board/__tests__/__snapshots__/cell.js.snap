// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<cell> should show a NUMBER1 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-1.png\\" src=\\"unsafe:number-1.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER2 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-2.png\\" src=\\"unsafe:number-2.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER3 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-3.png\\" src=\\"unsafe:number-3.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER4 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-4.png\\" src=\\"unsafe:number-4.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER5 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-5.png\\" src=\\"unsafe:number-5.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER6 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-6.png\\" src=\\"unsafe:number-6.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER7 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-7.png\\" src=\\"unsafe:number-7.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a NUMBER8 spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"number-8.png\\" src=\\"unsafe:number-8.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a flagged spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged --><span ng-if=\\"!Cell.spot.isRevealed &amp;&amp; Cell.spot.isFlagged\\" class=\\"ng-scope\\">
  <img ng-src=\\"flag-mine.png\\" src=\\"unsafe:flag-mine.png\\">
</span><!-- end ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show a mine spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"mine.png\\" src=\\"unsafe:mine.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show an empty spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed --><span ng-if=\\"Cell.spot.isRevealed\\" class=\\"ng-scope\\">
  <img ng-src=\\"empty.png\\" src=\\"unsafe:empty.png\\">
</span><!-- end ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;

exports[`<cell> should show an unrevealed spot 1`] = `
"<!-- ngIf: Cell.spot.isRevealed -->

<!-- ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged --><span ng-if=\\"!Cell.spot.isRevealed &amp;&amp; !Cell.spot.isFlagged\\" class=\\"ng-scope\\">
  <img ng-src=\\"covered.png\\" src=\\"unsafe:covered.png\\">
</span><!-- end ngIf: !Cell.spot.isRevealed && !Cell.spot.isFlagged -->

<!-- ngIf: !Cell.spot.isRevealed && Cell.spot.isFlagged -->

"
`;
